index.bs

```diff
@@ -10467,177 +10467,195 @@ asynchronous sections can't be reordered.
 <h3 id="rendering-loop">
 Rendering an audio graph</h3>
 
-Rendering an audio graph is done in blocks of 128 samples-frames. A
+Rendering an audio graph is done by processing blocks of 128 samples-frames. A
 block of 128 samples-frames is called a <dfn>render quantum</dfn>.
 
-Operations that happen <dfn>atomically</dfn> on a
-given thread can only be executed when no other [=Atomically|atomic=]
-operation is running on another thread.
+Operations that happen <dfn>atomically</dfn> on a given thread can only be
+executed when no other [=Atomically|atomic=] operation is running on another
+thread.
 
-The algorithm for rendering a block of audio from an
-{{AudioContext}} <em>G</em> with a <a href="#control-message-queue">control message queue</a> <em>Q</em> is as
-follows.
+The algorithm for rendering a block of audio from an {{AudioContext}} <em>G</em>
+with a <a href="#control-message-queue">control message queue</a> <em>Q</em> is
+comprised of 3 steps:
+	1. Process the control message queue.
+	2. Process a render quantum.
+	3. Perform a microtask check point.
 
-If the algorithm returns true then it MUST be executed again in the
-future, to render the next block of audio. Else, the <a>rendering
-thread</a> yields and the processing stops. The <a>control thread</a>
-can restart executing this algoritm if needed.
+The order of these steps cannot be altered or interrupted. If the second step
+returns true then it MUST be executed again in the future, to render the
```

Comment:
This is messy.  Step 2 says "process a render quantum".  There's no concept of returning anything here and for all nodes except a worklet, there's nothing to look at.